pkg/operator/controller/names.go

```diff
@@ -287,8 +288,12 @@ func ServiceMeshSubscriptionName() types.NamespacedName {
 // is named using the Gateway's name, listener's hashed host name, and the
 // suffix "-wildcard".
 func GatewayDNSRecordName(gateway *gatewayapiv1beta1.Gateway, host string) types.NamespacedName {
+	name := fmt.Sprintf("%s-%s", gateway.Name, util.Hash(host))
+	if strings.HasPrefix(host, "*.") {
+		name = name + "-wildcard"
+	}
 	return types.NamespacedName{
 		Namespace: gateway.Namespace,
-		Name:      fmt.Sprintf("%s-%s-wildcard", gateway.Name, util.Hash(host)),
+		Name:      name,
```

Comment:
We can't change this, this is a standard way of naming the DNSRecord, and it will always be a wildcard.